.pep8speaks.yml

```diff
@@ -16,4 +16,5 @@ pycodestyle:
         - E731,  # do not assign a lambda expression, use a def
         - C406,  # Unnecessary list literal - rewrite as a dict literal.
         - C408,  # Unnecessary dict call - rewrite as a literal.
-        - C409   # Unnecessary list passed to tuple() - rewrite as a tuple literal.
+        - C409,  # Unnecessary list passed to tuple() - rewrite as a tuple literal.
+        - F821   # undefined name (temporarily for type hints)
```

Comment:
I'm not actually sure if we want to do this since it would affect all undefined names (not just types) but I couldn't find a way to disambiguate save updating all of the type comments to have an ignore directive. If we wanted to go that route would probably be easier to move forward with some pieces of #25622 but open to thoughts

Reply:
That's unfortunate. I think it's probably better to pin to an older flake8 until #25622 is sorted out.

Reply:
FWIW, I don't see this error in the log:
```
./pandas/tests/indexes/interval/test_interval_tree.py(162,20): error F632: use ==/!= to compare str, bytes, and int literals
./pandas/tests/frame/test_alter_axes.py(204,32): error F632: use ==/!= to compare str, bytes, and int literals
./pandas/tests/frame/test_alter_axes.py(204,51): error F632: use ==/!= to compare str, bytes, and int literals
./pandas/tests/frame/test_alter_axes.py(1275,13): error E117: over-indented (comment)
./pandas/tests/io/test_pytables.py(2389,13): error E117: over-indented
./pandas/tests/series/test_rank.py(424,9): error E117: over-indented
./pandas/tests/series/test_rank.py(442,9): error E117: over-indented
./pandas/tests/series/test_rank.py(460,9): error E117: over-indented
./pandas/tests/series/test_rank.py(479,9): error E117: over-indented
./pandas/tests/series/test_rank.py(497,9): error E117: over-indented
./pandas/tests/series/test_rank.py(506,9): error E117: over-indented (comment)
```

The other changes in this PR seem worthwhile to me, irrespective of what happens with MyPy/F821.

Reply:
F821 is a doozy, is one of the checks that frequently catches actual bugs.  I’d very much prefer not to ignore it.

Reply:
@jbrockmendel for sure. The solution here is tucked away in #25622 so I'll be splitting that out when the timing is right. Waiting for official drop of Py2 in CI before that can happen